### PR TITLE
Fix Lint Warnings

### PR DESCRIPTION
**Pull request by bilal-alsharifi**
Fixes #1306 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [x] I have run the unit tests with this PR
- [x] I have tested this PR against Core and verified behavior (if applicable, if not applicable, explain why below).
- [x] I have tested Android, Java SE

### Summary
This PR fixes some of the Lint warnings in `android`, `javaSE`, `hello sdl java`, and `hello sdl android` projects

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: android/sdl_android/src/main/java/com/smartdevicelink/transport/utl/SdlDeviceListener.java [hunk 1]
```diff
@@ -197,7 +197,7 @@ public static void setSDLConnectedStatus(Context context, String address, boolea
                 }
                 SharedPreferences.Editor editor = preferences.edit();
                 editor.putBoolean(address, hasSDLConnected);
-                editor.commit();
+                editor.apply();
```
Comment (joeygrover) [inline android/sdl_android/src/main/java/com/smartdevicelink/transport/utl/SdlDeviceListener.java:200]: We specifically want to wait for this commit to happen; apply will return immediately.

File: base/src/main/java/com/smartdevicelink/transport/SiphonServer.java [hunk 2]
```diff
@@ -234,14 +234,11 @@ private static Boolean sendSiphonData(SiphonDataType direction, byte[] msgBytes,
...
 		long currentDateTime = System.currentTimeMillis();
-		Integer deltaTimeMills = null;
+		Integer deltaTimeMills;
 			
 		deltaTimeMills = (int)(currentDateTime - m_startTimeStamp);
 		
 		switch(direction) {
-			case fromSdl:
-				siphonDataTypeIndicator = 0x00;
-				break;
```
Comment (RHenigan) [inline base/src/main/java/com/smartdevicelink/transport/SiphonServer.java:244]: Why was this removed?

**Review by RHenigan — CHANGES_REQUESTED** [review]

File: base/src/main/java/com/smartdevicelink/transport/SiphonServer.java (hunk 2 above)
Comment (bilal-alsharifi) [reply base/src/main/java/com/smartdevicelink/transport/SiphonServer.java:244]: Because it is covered in the default entry. But I will add it back if you think it makes it less readable.

Comment (RHenigan) [reply base/src/main/java/com/smartdevicelink/transport/SiphonServer.java:244]: I think it is fine to leave it out

**Review by RHenigan — APPROVED** [review]
